### PR TITLE
Material: Make the frozen mode more user friendly

### PR DESCRIPTION
**Pull request by Popov72**
See https://forum.babylonjs.com/t/how-to-rotate-texture-of-frozen-material/37430

The PR allows you to simply call `material.markDirty(true)` to update a material even when in frozen mode, and even if the mesh(es) using the material are out of view at the time of the call.

Currently, it is difficult to update a frozen material, even using the **call unfreeze / make changes / call freeze** paradigm, because the latter has be postponed after we are sure that the material is actually changed, i.e., until after the new effect (if a new effect is created) is ready, which can be difficult to determine: it may take several frames for the effect to be ready once compiling starts, and compiling won't start if all the meshes that use that material are out of sight.

### CODE REVIEW

**Comment by bjsplat** [comment]
Please make sure to label your PR with "bug", "new feature" or "breaking change" label(s).
To prevent this PR from going to the changelog marked it with the "skip changelog" label.

**Comment by bjsplat** [comment]
Please make sure to label your PR with "bug", "new feature" or "breaking change" label(s).
To prevent this PR from going to the changelog marked it with the "skip changelog" label.

**Comment by bjsplat** [comment]
Snapshot stored with reference name:
refs/pull/13437/merge

Test environment: 
https://babylonsnapshots.z22.web.core.windows.net/refs/pull/13437/merge/index.html

To test a playground add it to the URL, for example:

https://babylonsnapshots.z22.web.core.windows.net/refs/pull/13437/merge/index.html#WGZLGJ#4600

Links to test babylon tools with this snapshot:

https://playground.babylonjs.com/?snapshot=refs/pull/13437/merge
https://sandbox.babylonjs.com/?snapshot=refs/pull/13437/merge
https://gui.babylonjs.com/?snapshot=refs/pull/13437/merge
https://nme.babylonjs.com/?snapshot=refs/pull/13437/merge

To test the snapshot in the playground with a playground ID add it after the snapshot query string:

https://playground.babylonjs.com/?snapshot=refs/pull/13437/merge#BCU1XR#0

File: packages/dev/core/src/Materials/material.ts [hunk 1]
```diff
@@ -1125,8 +1125,9 @@ export class Material implements IAnimatable, IClipPlanesHolder {
 
     /**
      * Marks the material to indicate that it needs to be re-calculated
+     * @param forceDirtyfyAll - Forces the material to be marked as dirty for all components (same as this.markAsDirty(Material.AllDirtyFlag))
      */
-    public markDirty(): void {
+    public markDirty(forceDirtyfyAll = false): void {
```
Comment (RaananW) [inline packages/dev/core/src/Materials/material.ts:1130]: Did you mean to call the flag this way?

**Review by RaananW — APPROVED** [review]
LGTM! just a naming question.

**Review by sebavan — COMMENTED** [review]
just wondering how we could prevent the extra checks ? but not sure it is possible ?

Maybe a helper to refreeze when rendered using observable ???

**Comment by bjsplat** [comment]
Visualization tests for webgl2 have failed. If some tests failed because the snapshots do not match, the report can be found at

https://babylonsnapshots.z22.web.core.windows.net/refs/pull/13437/merge/testResults/webgl2/index.html

If tests were successful afterwards, this report might not be available anymore.

**Comment by bjsplat** [comment]
Visualization tests for webgl1 have failed. If some tests failed because the snapshots do not match, the report can be found at

https://babylonsnapshots.z22.web.core.windows.net/refs/pull/13437/merge/testResults/webgl1/index.html

If tests were successful afterwards, this report might not be available anymore.

**Pull request merged**